File: app/src/main/java/com/mobile/rxjava2andretrofit2/main/MainActivity.java
package com.mobile.rxjava2andretrofit2.main;

import android.os.Bundle;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.fragment.app.Fragment;

import com.gyf.immersionbar.ImmersionBar;
import com.mobile.rxjava2andretrofit2.R;
import com.mobile.rxjava2andretrofit2.base.BaseMvpAppActivity;
import com.mobile.rxjava2andretrofit2.base.IBaseView;
import com.mobile.rxjava2andretrofit2.custom_view.LazyViewPager;
import com.mobile.rxjava2andretrofit2.custom_view.MineLazyViewPager;
import com.mobile.rxjava2andretrofit2.first_page.FirstPageFragment;
import com.mobile.rxjava2andretrofit2.kotlin.mine.MineFragment;
import com.mobile.rxjava2andretrofit2.kotlin.resources_show.ResourceCenterFragment;
import com.mobile.rxjava2andretrofit2.main.adapter.TabFragmentPagerAdapter;
import com.mobile.rxjava2andretrofit2.main.presenter.MainPresenterImpl;
import com.mobile.rxjava2andretrofit2.main.view.IMainView;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.OnClick;

public class MainActivity extends BaseMvpAppActivity<IBaseView, MainPresenterImpl>
        implements IMainView {

    private static final String TAG = "MainActivity";
    @BindView(R.id.mine_view_pager)
    MineLazyViewPager mineViewPager;
    @BindView(R.id.layout_bottom)
    LinearLayout layoutBottom;
    @BindView(R.id.tev_first_page)
    TextView tevFirstPage;
    @BindView(R.id.tev_resource_center)
    TextView tevResourceCenter;
    @BindView(R.id.tev_mine)
    TextView tevMine;

    private List<Fragment> fragmentList = new ArrayList<>();
    private TabFragmentPagerAdapter fragmentPagerAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    protected int initLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    protected void initData() {
        fragmentList.add(new FirstPageFragment());
        fragmentList.add(new ResourceCenterFragment());
        fragmentList.add(new MineFragment());
    }

    @Override
    protected void initViews() {
//        ImmersionBar.with(MainActivity.this)
//                .keyboardEnable(false)
//                .statusBarDarkFont(false)
//                .statusBarColor(R.color.color_FFE066FF)
//                .navigationBarColor(R.color.color_FFE066FF).init();
        addContentView(loadView, layoutParams);

        fragmentPagerAdapter = new TabFragmentPagerAdapter(getSupportFragmentManager(), fragmentList);
        mineViewPager.setAdapter(fragmentPagerAdapter);
        mineViewPager.setOnPageChangeListener(new LazyViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

            }

            @Override
            public void onPageSelected(int position) {
                switch (position) {
                    case 0:
                        mineViewPager.setCurrentItem(0);
                        tevFirstPage.setTextColor(getResources().getColor(R.color.color_FFE066FF));
                        tevResourceCenter.setTextColor(getResources().getColor(R.color.color_FF999999));
                        tevMine.setTextColor(getResources().getColor(R.color.color_FF999999));

                        ImmersionBar.with(MainActivity.this)
                                .keyboardEnable(false)
                                .statusBarDarkFont(false)
                                .statusBarColor(R.color.color_FFE066FF)
                                .navigationBarColor(R.color.color_FFE066FF).init();
                        layoutBottom.setBackgroundColor(getResources().getColor(R.color.white));
                        break;
                    case 1:
                        mineViewPager.setCurrentItem(1);
                        tevFirstPage.setTextColor(getResources().getColor(R.color.color_FF999999));
                        tevResourceCenter.setTextColor(getResources().getColor(R.color.color_FFE066FF));
                        tevMine.setTextColor(getResources().getColor(R.color.color_FF999999));

                        ImmersionBar.with(MainActivity.this)
                                .keyboardEnable(false)
                                .statusBarDarkFont(false)
                                .statusBarColor(R.color.color_FF198CFF)
                                .navigationBarColor(R.color.color_FF198CFF).init();
                        layoutBottom.setBackgroundColor(getResources().getColor(R.color.white));
                        break;
                    case 2:
                        mineViewPager.setCurrentItem(2);
                        tevFirstPage.setTextColor(getResources().getColor(R.color.color_FF999999));
                        tevResourceCenter.setTextColor(getResources().getColor(R.color.color_FF999999));
                        tevMine.setTextColor(getResources().getColor(R.color.color_FFE066FF));

                        ImmersionBar.with(MainActivity.this)
                                .keyboardEnable(false)
                                .statusBarDarkFont(false)
                                .statusBarColor(R.color.color_FFE066FF)
                                .navigationBarColor(R.color.color_FFE066FF).init();
                        layoutBottom.setBackgroundColor(getResources().getColor(R.color.black));
                        break;
                    default:
                        break;
                }
            }

            @Override
            public void onPageScrollStateChanged(int state) {

            }
        });
    }

    @Override
    protected void initLoadData() {
        mineViewPager.setCurrentItem(0);
        tevFirstPage.setTextColor(getResources().getColor(R.color.color_FFE066FF));
        tevResourceCenter.setTextColor(getResources().getColor(R.color.color_FF999999));
        tevMine.setTextColor(getResources().getColor(R.color.color_FF999999));

        ImmersionBar.with(MainActivity.this)
                .keyboardEnable(false)
                .statusBarDarkFont(false)
                .statusBarColor(R.color.color_FFE066FF)
                .navigationBarColor(R.color.color_FFE066FF).init();
    }

    @Override
    protected MainPresenterImpl attachPresenter() {
        return new MainPresenterImpl(this);
    }

    @Override
    public void showLoading() {
        if (loadView != null && !loadView.isShown()) {
            loadView.setVisibility(View.VISIBLE);
            loadView.start();
        }
    }

    @Override
    public void hideLoading() {
        if (loadView != null && loadView.isShown()) {
            loadView.stop();
            loadView.setVisibility(View.GONE);
        }
    }

    @Override
    public void mainDataSuccess(String success) {
        showToast(success, true);
    }

    @Override
    public void mainDataError(String error) {
        showToast(error, true);
    }

    @OnClick({R.id.tev_first_page, R.id.tev_resource_center, R.id.tev_mine})
    public void onViewClicked(View view) {
        switch (view.getId()) {
            case R.id.tev_first_page:
                mineViewPager.setCurrentItem(0);
                tevFirstPage.setTextColor(getResources().getColor(R.color.color_FFE066FF));
                tevResourceCenter.setTextColor(getResources().getColor(R.color.color_FF999999));
                tevMine.setTextColor(getResources().getColor(R.color.color_FF999999));
                break;
            case R.id.tev_resource_center:
                mineViewPager.setCurrentItem(1);
                tevFirstPage.setTextColor(getResources().getColor(R.color.color_FF999999));
                tevResourceCenter.setTextColor(getResources().getColor(R.color.color_FFE066FF));
                tevMine.setTextColor(getResources().getColor(R.color.color_FF999999));
                break;
            case R.id.tev_mine:
                mineViewPager.setCurrentItem(2);
                tevFirstPage.setTextColor(getResources().getColor(R.color.color_FF999999));
                tevResourceCenter.setTextColor(getResources().getColor(R.color.color_FF999999));
                tevMine.setTextColor(getResources().getColor(R.color.color_FFE066FF));
                break;
        }
    }
}

File: main_module/src/main/java/com/phone/main_module/main/MainActivity.kt
package com.phone.main_module.main

import android.view.View
import android.widget.LinearLayout
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.alibaba.android.arouter.launcher.ARouter
import com.phone.common_library.adapter.TabFragmentStatePagerAdapter
import com.phone.common_library.base.BaseMvpRxAppActivity
import com.phone.common_library.base.IBaseView
import com.phone.common_library.custom_view.LazyViewPager
import com.phone.common_library.custom_view.MineLazyViewPager
import com.phone.common_library.manager.ResourcesManager
import com.phone.main_module.BuildConfig
import com.phone.main_module.R
import com.phone.main_module.main.presenter.MainPresenterImpl
import com.phone.main_module.main.view.IMainView

/**
 * author    : Urasaki
 * e-mail    : 1164688204@qq.com
 * date      :
 * introduce :
 */
class MainActivity : BaseMvpRxAppActivity<IBaseView, MainPresenterImpl>(), IMainView {

    private val TAG = MainActivity::class.java.simpleName
    private var tevPleaseAddComponents: TextView? = null
    private var layoutMain: LinearLayout? = null
    private var mineViewPager: MineLazyViewPager? = null
    private var layoutBottom: LinearLayout? = null
    private var tevFirstPage: TextView? = null
    private var tevProject: TextView? = null
    private var tevSquare: TextView? = null
    private var tevResourceCenter: TextView? = null
    private var tevMine: TextView? = null

    private val fragmentList: MutableList<Fragment> = ArrayList()

    override fun initLayoutId(): Int {
        return R.layout.activity_main
    }

    override fun initData() {
        if (!BuildConfig.IS_MODULE) {
            //Jump with parameters
            val firstPageFragment = ARouter.getInstance()
                .build("/first_page_module/first_page") //                .withString("max_behot_time", (System.currentTimeMillis() / 1000) + "")
                .navigation() as Fragment
            fragmentList.add(firstPageFragment)
            //Jump with parameters
            val projectFragment = ARouter.getInstance()
                .build("/project_module/project") //                .withString("max_behot_time", (System.currentTimeMillis() / 1000) + "")
                .navigation() as Fragment
            fragmentList.add(projectFragment)
            //Jump with parameters
            val squareFragment = ARouter.getInstance()
                .build("/square_module/square") //                .withString("max_behot_time", (System.currentTimeMillis() / 1000) + "")
                .navigation() as Fragment
            fragmentList.add(squareFragment)
            //Jump with parameters
            val resourceFragment = ARouter.getInstance()
                .build("/resource_module/resource") //                .withString("max_behot_time", (System.currentTimeMillis() / 1000) + "")
                .navigation() as Fragment
            fragmentList.add(resourceFragment)
            //Jump with parameters
            val mineFragment = ARouter.getInstance()
                .build("/mine_module/mine") //                .withString("max_behot_time", (System.currentTimeMillis() / 1000) + "")
                .navigation() as Fragment
            fragmentList.add(mineFragment)
        }
    }

    override fun initViews() {
        tevPleaseAddComponents = findViewById<View>(R.id.tev_please_add_components) as TextView
        layoutMain = findViewById<View>(R.id.layout_main) as LinearLayout
        mineViewPager = findViewById<View>(R.id.mine_view_pager) as MineLazyViewPager
        layoutBottom = findViewById<View>(R.id.layout_bottom) as LinearLayout
        tevFirstPage = findViewById<View>(R.id.tev_first_page) as TextView
        tevProject = findViewById<View>(R.id.tev_project) as TextView
        tevSquare = findViewById<View>(R.id.tev_square) as TextView
        tevResourceCenter = findViewById<View>(R.id.tev_resource_center) as TextView
        tevMine = findViewById<View>(R.id.tev_mine) as TextView
        val fragmentStatePagerAdapter = TabFragmentStatePagerAdapter(
            supportFragmentManager, fragmentList
        )
        mineViewPager?.adapter = fragmentStatePagerAdapter
        mineViewPager?.setOnPageChangeListener(object : LazyViewPager.OnPageChangeListener {
            override fun onPageScrolled(
                position: Int,
                positionOffset: Float,
                positionOffsetPixels: Int
            ) {
            }

            override fun onPageSelected(position: Int) {
                resetTabData(position)
            }

            override fun onPageScrollStateChanged(state: Int) {}
        })
        tevFirstPage?.setOnClickListener { v: View? ->
            resetTabData(
                0
            )
        }
        tevProject?.setOnClickListener { v: View? ->
            resetTabData(
                1
            )
        }
        tevSquare?.setOnClickListener { v: View? ->
            resetTabData(
                2
            )
        }
        tevResourceCenter?.setOnClickListener { v: View? ->
            resetTabData(
                3
            )
        }
        tevMine?.setOnClickListener { v: View? ->
            resetTabData(
                4
            )
        }
    }

    private fun resetTabData(position: Int) {
        mineViewPager?.currentItem = position
        tevFirstPage?.setTextColor(ResourcesManager.getColor(R.color.color_FF999999))
        tevProject?.setTextColor(ResourcesManager.getColor(R.color.color_FF999999))
        tevSquare?.setTextColor(ResourcesManager.getColor(R.color.color_FF999999))
        tevResourceCenter?.setTextColor(ResourcesManager.getColor(R.color.color_FF999999))
        tevMine?.setTextColor(ResourcesManager.getColor(R.color.color_FF999999))
        setToolbar(false, R.color.color_FF198CFF)
        when (position) {
            0 -> {
                tevFirstPage?.setTextColor(ResourcesManager.getColor(R.color.color_FFE066FF))
                layoutBottom?.setBackgroundColor(ResourcesManager.getColor(R.color.black))
            }
            1 -> {
                tevProject?.setTextColor(ResourcesManager.getColor(R.color.color_FFE066FF))
                layoutBottom?.setBackgroundColor(ResourcesManager.getColor(R.color.white))
            }
            2 -> {
                tevSquare?.setTextColor(ResourcesManager.getColor(R.color.color_FFE066FF))
                layoutBottom?.setBackgroundColor(ResourcesManager.getColor(R.color.white))
            }
            3 -> {
                tevResourceCenter?.setTextColor(ResourcesManager.getColor(R.color.color_FFE066FF))
                layoutBottom?.setBackgroundColor(ResourcesManager.getColor(R.color.white))
            }
            4 -> {
                tevMine?.setTextColor(ResourcesManager.getColor(R.color.color_FFE066FF))
                layoutBottom?.setBackgroundColor(ResourcesManager.getColor(R.color.black))
            }
            else -> {
                tevFirstPage?.setTextColor(ResourcesManager.getColor(R.color.color_FFE066FF))
                layoutBottom?.setBackgroundColor(ResourcesManager.getColor(R.color.black))
            }
        }
    }

    override fun initLoadData() {
        if (!BuildConfig.IS_MODULE) {
            tevPleaseAddComponents?.visibility = View.GONE
            layoutMain?.visibility = View.VISIBLE
            mineViewPager?.currentItem = 0
            tevFirstPage?.setTextColor(ResourcesManager.getColor(R.color.color_FFE066FF))
            tevProject?.setTextColor(ResourcesManager.getColor(R.color.color_FF999999))
            tevSquare?.setTextColor(ResourcesManager.getColor(R.color.color_FF999999))
            tevResourceCenter?.setTextColor(ResourcesManager.getColor(R.color.color_FF999999))
            tevMine?.setTextColor(ResourcesManager.getColor(R.color.color_FF999999))
            setToolbar(false, R.color.color_FF198CFF)
        } else {
            tevPleaseAddComponents?.visibility = View.VISIBLE
            layoutMain?.visibility = View.GONE
        }
    }

    override fun attachPresenter(): MainPresenterImpl {
        return MainPresenterImpl(this)
    }

    override fun showLoading() {
        if (!mLoadView.isShown) {
            mLoadView.visibility = View.VISIBLE
            mLoadView.start()
        }
    }

    override fun hideLoading() {
        if (mLoadView.isShown) {
            mLoadView.stop()
            mLoadView.visibility = View.GONE
        }
    }

    override fun mainDataSuccess(success: String) {
        showToast(success, true)
    }

    override fun mainDataError(error: String) {
        showToast(error, true)
    }

}